### PR TITLE
Explicitly call optional validator first in validateSchema function

### PR DESCRIPTION
**Pull request by artemsmirnov**

### CODE REVIEW

File: lib/express_validator.js [hunk 1]
```diff
@@ -342,6 +342,12 @@ function validateSchema(schema, req, loc, options) {
     var validator = new ValidatorChain(param, null, req, currentLoc, options);
     var paramErrorMessage = schema[param].errorMessage;
 
+    // Since javascript object keys have order we explicitly call optional validator very first
```
Comment (rustybailey) [inline lib/express_validator.js]: Did you mean to say "have no order" here?

File: lib/express_validator.js [hunk 2]
```diff
@@ -360,6 +366,13 @@ function validateSchema(schema, req, loc, options) {
         continue;
       }
 
+      if (methodName === 'optional') {
+        /**
+         * We already done this before
```
Comment (rustybailey) [inline lib/express_validator.js]: Change this comment to something like: "We explicitly do optional first if it is present, so skip it here."

**Comment by gustavohenke** [comment]
ping @artemsmirnov, do you mind checking the reviews made by Rusty?

**Comment by gustavohenke** [comment]
I'll close this PR, as it has been superseded by #282.

**Pull request closed without merging**